netty/src/main/java/io/grpc/transport/netty/NettyServerStream.java

```diff
@@ -85,8 +95,13 @@ protected void inboundDeliveryPaused() {
 
   @Override
   protected void internalSendHeaders(Metadata.Headers headers) {
-    channel.writeAndFlush(new SendResponseHeadersCommand(id(),
-        Utils.convertServerHeaders(headers), false));
+    // No need to flush headers yet if the server is required to respond with one message.
```

Comment:
Same comment as the client case ... why would we ever want to flush when we're just sending headers?